src/cdk/table/text-column.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  Component,
+  Inject,
+  InjectionToken,
+  Input,
+  OnDestroy,
+  OnInit,
+  Optional,
+  ViewChild,
+  ChangeDetectionStrategy,
+  ViewEncapsulation
+} from '@angular/core';
+import {CdkColumnDef} from './cell';
+import {CdkTable} from './table';
+
+/** Configurable options for `CdkTextColumn`. */
+export interface TextColumnOptions {
+  /**
+   * Default function that provides the header text based on the column name if a header
+   * text is not provided.
+   */
+  defaultHeaderTextTransformation?: (name: string) => string;
+
+  /** Default data accessor to use if one is not provided. */
+  defaultDataAccessor?: (data: any, name: string) => string;
+}
+
+/** Injection token that can be used to specify the text column options. */
+export const TEXT_COLUMN_OPTIONS = new InjectionToken<TextColumnOptions>('text-column-options');
+
+/**
+ * Column that simply shows text content for the header and row cells. Assumes that the table
+ * is using the native table implementation (`<table>`).
+ *
+ * By default, the name of this column will be the header text and data property accessor.
```

Comment:
I'd prefer to lean towards leaving the behavior as is, but make sure to mention in the documentation that this may come up. I suspect it is very rare for users to have minified properties on their table data and the added API requirement will seem very redundant for everyone else.

We do something similar in the `MatTableDataSource` where, by default, the sort accessor (`sortingDataAccessor`) uses property access by the sort header ID which will have the same minification problems. I haven't seen any issues come in regarding confusion about it.

Happy to chat more about it in case I misunderstand how often this would occur

Reply:
Okay- can you add the documentation in this PR?